addons/survey/controllers/main.py

```diff
@@ -25,21 +26,35 @@ class Survey(http.Controller):
     # ACCESS
     # ------------------------------------------------------------
 
-    def _fetch_from_access_token(self, survey_token, answer_token):
+    def fetch_answer(self, survey_token, answer_token, survey_id=False, answer_id=False):
```

Comment:
As the compatibility routes should be giving us the id, can't we remove the compatibility code from these methods?

Make ids required fields, remove "if not id" blocks etc...